x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -531,6 +533,166 @@ class AgentPolicyService {
     };
   }
 
+  public async bulkUpdate(
```

Comment:
Wow - ok. this is a full "bulk" update of agent policies. When I initially provided feedback I was thinking that we would introduce a `bulkUpdateTamperProtection()` method that would be processed in a much more efficient way (aka: not do all of the normal agent policy update logic) and just be able to update the protection on agent policies.  You can still keep it if Fleet feels it can be useful, but I was hopping that we had something that can execute faster by limiting what can be updated.

I'll still provide feedback below, but if you change direction it may no longer apply

Reply:
@kpollich @juliaElastic What do you think? :)

Reply:
@kpollich @juliaElastic

I'm okay with changing the approach, in the end this implementation was supposed to solve issues instead of introducing new ones. Thanks for the feedback! :) 

The way I see this narrower API for just agent tamper protection:
1. Find all the policies with `is_protected: true` 
2. Do `SO.bulkUpdate([{..., is_protected: false}])` with bumped `revision` and `updated_at`
3. Call for a deploy on all affected policies. I believe you've mentioned Fleet's actions offline to achieve that, I see that AgentPolicy exposes `deployPolicies` yet it still does a pMap of `agentPolicy.getFullAgentPolicy`. I guess this is not what you meant? :)

Reply:
> Call for a deploy on all affected policies. I believe you've mentioned Fleet's actions offline to achieve that, I see that AgentPolicy exposes deployPolicies is that it?

I don't think we need to reach for actions in this context, but you can see an example of what I mean in Fleet's `bulk_actions_resolver` module. 

For this implementation, I think we can something similar to what we do in the uninstall token service: batching the policies into chunks and looping through those chunks sequentially, e.g. 

https://github.com/elastic/kibana/blob/1c7d0891623d44ef4b85fb84b891dc59fd5186aa/x-pack/plugins/fleet/server/services/security/uninstall_token_service/index.ts#L416-L425

The `pMap` approach is probably okay too, compared to the `asyncForEach` - but if we're not exposing this method as a generic way to bulk update policies then I think we are okay to just batch process the updates in this service.